api/src/test/java/org/apache/iceberg/util/TestExceptionUtil.java

```diff
@@ -37,178 +34,110 @@ private CustomCheckedException(String message) {
   @Test
   public void testRunSafely() {
     CustomCheckedException exc = new CustomCheckedException("test");
-    try {
```

Comment:
I have to admit that the original test code was quite difficult to read, so I tried to simplify it to the bare minimum